flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -754,15 +754,16 @@ private boolean allocateResource(PendingTaskManager pendingTaskManager) {
             return false;
         }
 
-        if (!resourceAllocator.isSupported()
-                || !resourceAllocator.allocateResource(
-                        WorkerResourceSpec.fromTotalResourceProfile(
-                                pendingTaskManager.getTotalResourceProfile(),
-                                pendingTaskManager.getNumSlots()))) {
+        if (!resourceAllocator.isSupported()) {
```

Comment:
It would be better to move this check to before the max total resource check, because this one is cheaper.